.github/scripts/generate_ci_workflows.py

```diff
@@ -25,13 +25,15 @@ def PyTorchWindowsWorkflow(
     test_runner_type: str,
     cuda_version: str,
     on_pull_request: bool = False,
+    only_build_on_pull_request: bool = False,
     num_test_shards: int = 1,
 ) -> PyTorchWorkflow:
     return {
         "build_environment": build_environment,
         "test_runner_type": test_runner_type,
         "cuda_version": cuda_version,
         "on_pull_request": on_pull_request,
+        "only_build_on_pull_request": only_build_on_pull_request and on_pull_request,
```

Comment:
There should never be a scenario where `only_build_on_pull_request=True` and `on_pull_request=False`